File: commerce_app/lib/models/merchant.dart
class Merchant {
  final int id;
  final int ownerId;
  final String name;
  final String? description;
  final String? email;
  final String? phone;
  final String? logoUrl;
  final String? address;
  final String? city;
  final String? country;
  final int? categoryId;
  final String status;
  final bool? verified;
  final String? createdAt;
  final double? latitude;
  final double? longitude;
  final String? openingHours;

  Merchant({
    required this.id,
    required this.ownerId,
    required this.name,
    this.description,
    this.email,
    this.phone,
    this.logoUrl,
    this.address,
    this.city,
    this.country,
    this.categoryId,
    required this.status,
    this.verified,
    this.createdAt,
    this.latitude,
    this.longitude,
    this.openingHours,
  });

  factory Merchant.fromJson(Map<String, dynamic> json) {
    return Merchant(
      id: (json['id'] as num).toInt(),
      ownerId: (json['ownerId'] as num).toInt(),
      name: json['name'] as String? ?? '',
      description: json['description'] as String?,
      email: json['email'] as String?,
      phone: json['phone'] as String?,
      logoUrl: json['logoUrl'] as String?,
      address: json['address'] as String?,
      city: json['city'] as String?,
      country: json['country'] as String?,
      categoryId: (json['categoryId'] as num?)?.toInt(),
      status: json['status'] as String? ?? 'PENDING',
      verified: json['verified'] as bool?,
      createdAt: json['createdAt'] as String?,
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
      openingHours: json['openingHours'] as String?,
    );
  }
}

File: backend/src/main/java/com/campuspass/backend/model/Merchant.java
package com.campuspass.backend.model;

import com.campuspass.backend.model.enums.MerchantStatus;
import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "merchants")
public class Merchant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long ownerId;

    @Column(nullable = false)
    private String name;
    private String description;
    private String email;
    private String phone;
    private String website;
    private String logoUrl;
    private String coverImage;

    private Long categoryId;
    private String address;
    private String city;
    /** Quartier / zone (ex: Ouaga 2000, Zone du Bois). */
    private String neighborhood;
    private String country;
    private Double latitude;
    private Double longitude;

    /** Note moyenne (0-5) basée sur les avis. */
    private Double rating;

    /** Horaires d'ouverture (format libre, ex: "Lun-Dim 08h00 - 22h00"). */
    private String openingHours;

    private Boolean verified = false;
    private LocalDateTime verificationDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private MerchantStatus status = MerchantStatus.PENDING;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ownerId", insertable = false, updatable = false)
    private User owner;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "categoryId", insertable = false, updatable = false)
    private Category category;

    public Merchant() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getOwnerId() { return ownerId; }
    public void setOwnerId(Long ownerId) { this.ownerId = ownerId; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }
    public String getEmail() { return email; }
    public void setEmail(String email) { this.email = email; }
    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; }
    public String getWebsite() { return website; }
    public void setWebsite(String website) { this.website = website; }
    public String getLogoUrl() { return logoUrl; }
    public void setLogoUrl(String logoUrl) { this.logoUrl = logoUrl; }
    public String getCoverImage() { return coverImage; }
    public void setCoverImage(String coverImage) { this.coverImage = coverImage; }
    public Long getCategoryId() { return categoryId; }
    public void setCategoryId(Long categoryId) { this.categoryId = categoryId; }
    public String getAddress() { return address; }
    public void setAddress(String address) { this.address = address; }
    public String getCity() { return city; }
    public void setCity(String city) { this.city = city; }
    public String getCountry() { return country; }
    public void setCountry(String country) { this.country = country; }
    public String getNeighborhood() { return neighborhood; }
    public void setNeighborhood(String neighborhood) { this.neighborhood = neighborhood; }
    public Double getLatitude() { return latitude; }
    public void setLatitude(Double latitude) { this.latitude = latitude; }
    public Double getLongitude() { return longitude; }
    public void setLongitude(Double longitude) { this.longitude = longitude; }
    public String getOpeningHours() { return openingHours; }
    public void setOpeningHours(String openingHours) { this.openingHours = openingHours; }
    public Double getRating() { return rating; }
    public void setRating(Double rating) { this.rating = rating; }
    public Boolean getVerified() { return verified; }
    public void setVerified(Boolean verified) { this.verified = verified; }
    public LocalDateTime getVerificationDate() { return verificationDate; }
    public void setVerificationDate(LocalDateTime verificationDate) { this.verificationDate = verificationDate; }
    public MerchantStatus getStatus() { return status; }
    public void setStatus(MerchantStatus status) { this.status = status; }
    public LocalDateTime getCreatedAt() { return createdAt; }
    public void setCreatedAt(LocalDateTime createdAt) { this.createdAt = createdAt; }
    public LocalDateTime getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(LocalDateTime updatedAt) { this.updatedAt = updatedAt; }
}
